code/modules/mob/living/simple_animal/guardian/types/assassin.dm

```diff
@@ -1,100 +1,83 @@
 	var/stealthcooldown = 160
 	var/obj/screen/alert/canstealthalert
 	var/obj/screen/alert/instealthalert
 
-/mob/living/simple_animal/hostile/guardian/assassin/Initialize()
-	..()
-	stealthcooldown = 0
+/datum/guardian_abilities/assassin/Destroy()
+	return ..()
 
-/mob/living/simple_animal/hostile/guardian/assassin/Life()
-	. = ..()
-	updatestealthalert()
-	if(loc == summoner && toggle)
-		ToggleMode(0)
+/datum/guardian_abilities/assassin/handle_stats()
+	guardian.has_mode = TRUE
+	guardian.melee_damage_lower += 7
+	guardian.melee_damage_upper += 7
+	guardian.attacktext = "slashes"
+	guardian.attack_sound = 'sound/weapons/bladeslice.ogg'
+	guardian.toggle_button_type = /obj/screen/guardian/ToggleMode/Assassin
 
-/mob/living/simple_animal/hostile/guardian/assassin/Stat()
-	..()
-	if(statpanel("Status"))
-		if(stealthcooldown >= world.time)
-			stat(null, "Stealth Cooldown Remaining: [max(round((stealthcooldown - world.time)*0.1, 0.1), 0)] seconds")
 
-/mob/living/simple_animal/hostile/guardian/assassin/AttackingTarget()
-	. = ..()
-	if(.)
-		if(toggle && (isliving(target) || istype(target, /obj/structure/window) || istype(target, /obj/structure/grille)))
-			ToggleMode(1)
+/datum/guardian_abilities/assassin/proc/updatestealthalert()
+	if(stealthcooldown <= world.time)
+		if(toggle)
+			if(!instealthalert)
+				instealthalert = guardian.throw_alert("instealth", /obj/screen/alert/instealth)
+				guardian.clear_alert("canstealth")
+				canstealthalert = null
+		else
+			if(!canstealthalert)
+				canstealthalert = guardian.throw_alert("canstealth", /obj/screen/alert/canstealth)
+				guardian.clear_alert("instealth")
+				instealthalert = null
+	else
+		guardian.clear_alert("instealth")
+		instealthalert = null
+		guardian.clear_alert("canstealth")
+		canstealthalert = null
 
-/mob/living/simple_animal/hostile/guardian/assassin/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
-	. = ..()
-	if(. > 0 && toggle)
-		ToggleMode(1)
 
-/mob/living/simple_animal/hostile/guardian/assassin/Recall()
-	if(..() && toggle)
-		ToggleMode(0)
+/datum/guardian_abilities/assassin/life_act()
+	updatestealthalert()
+	if(guardian.loc == user && toggle)
+		guardian.ToggleMode(0)
 
-/mob/living/simple_animal/hostile/guardian/assassin/ToggleMode(forced = 0)
+/datum/guardian_abilities/assassin/handle_mode(forced = 0)
 	if(toggle)
-		melee_damage_lower = initial(melee_damage_lower)
-		melee_damage_upper = initial(melee_damage_upper)
-		armour_penetration = initial(armour_penetration)
-		obj_damage = initial(obj_damage)
-		environment_smash = initial(environment_smash)
-		alpha = initial(alpha)
+		guardian.melee_damage_lower = initial(guardian.melee_damage_lower)
+		guardian.melee_damage_upper = initial(guardian.melee_damage_upper)
+		guardian.armour_penetration = initial(guardian.armour_penetration)
+		guardian.obj_damage = initial(guardian.obj_damage)
+		guardian.environment_smash = initial(guardian.environment_smash)
+		guardian.alpha = initial(guardian.alpha)
```

Comment:
and how am i supposed to do that

if you're gonna tell me to do something 

either 1. tell me how to do it (preferred)
2. tell me WHY the current version does not work, so i can fix that specific mistake.

don't cop out if you're gonna review.